tools/clang/unittests/SPIRV/CodeGenSpirvTest.cpp

```diff
@@ -2236,6 +2236,10 @@ TEST_F(FileTest, RichDebugInfoFunction) {
   runFileTest("rich.debug.function.hlsl", Expect::Success,
               /*runValidation*/ false);
 }
+TEST_F(FileTest, RichDebugInfoFunctionParam) {
+  runFileTest("rich.debug.function.param.hlsl", Expect::Success,
+              /*runValidation*/ false);
+}
```

Comment:
I would also add one more test in which "foo" is the member function of a class.

Reply:
I tested a member function of a struct. Do you want to add tests for "class"?
I intentionally left the separation between class/struct, which is one of TODO (since it is not that high priority for now).

Reply:
No need for "class" right now. But, I don't see any CHECKs for function parameters of a function that is a member of a struct. 
`rich.debug.function.param.hlsl` does not include a struct. (Looking for `DebugLocalVariable` for function params of a struct member function).